jetty-util/src/main/java/org/eclipse/jetty/util/URIUtil.java

```diff
@@ -1305,6 +1305,28 @@ public static URI addPath(URI uri, String path)
         return URI.create(buf.toString());
     }
 
+    /**
+     * Combine two query strings into one. Each query string should not contain the beginning '?' character, but
+     * may contain multiple parameters separated by the '{@literal &}' character.
+     * @param query1 the first query string.
+     * @param query2 the second query string.
+     * @return the combination of the two query strings.
+     */
+    public static String combineQueryParams(String query1, String query2)
```

Comment:
a better name is `addQueries` as that matches the `addPaths` we already have.
To make it more efficient it should do:
```java
if (StringUtil.isEmpty(query1))
  return query2;
if (StringUtil.isEmpty(query2))
  return query1;
return query1 + '&' + query2;
```